components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/CarbonAppResource.java

```diff
@@ -56,18 +78,128 @@ public boolean invoke(MessageContext messageContext) {
+            if (fileName != null && fileName.endsWith(".car")) {
+                byte[] bytes = Base64.getDecoder().decode(fileElement.getText());
+                Path cAppDirectoryPath = Paths.get(Utils.getCarbonHome(), "repository", "deployment",
+                        "server", "carbonapps", fileName);
+                try {
+                    Files.write(cAppDirectoryPath, bytes);
+                } catch (IOException e) {
+                    isDeployedSuccesfully = false;
+                    unMovedCApps.append(fileName).append(", ");
+                    log.error(errorMessage + fileName, e);
+                }
+                log.info("Successfully added Carbon Application : " + fileName);
+            } else {
+                isDeployedSuccesfully = false;
+                unMovedCApps.append((fileName != null ? fileName : "filename: <null>")).append(", ");
+            }
+        }
+
+        if (isDeployedSuccesfully) {
+            jsonResponse.put(Constants.MESSAGE_JSON_ATTRIBUTE, "Successfully added Carbon Application(s)");
+            Utils.setJsonPayLoad(axisMsgCtx, jsonResponse);
+        } else {
+            Utils.setJsonPayLoad(axisMsgCtx, Utils.createJsonErrorObject(errorMessage + unMovedCApps));
+        }
+    }
+
+    private void handleDelete(MessageContext messageContext, org.apache.axis2.context.MessageContext axisMsgCtx) {
+        String cAppNamePattern = Utils.getPathParameter(messageContext, CAPP_NAME_PATTERN);
+        JSONObject jsonResponse = new JSONObject();
+        if (!Objects.isNull(cAppNamePattern)) {
+            try {
+                String cAppsDirectoryPath = Paths.get(
+                        Utils.getCarbonHome(), "repository", "deployment", "server", "carbonapps").toString();
+
+                // List deployed CApps which has downloaded CApp name prefix
+                File carbonAppsDirectory = new File(cAppsDirectoryPath);
+                File[] existingCApps = carbonAppsDirectory.listFiles(new FilenameFilter() {
+                    public boolean accept(File dir, String name) {
+                        return name.contains(cAppNamePattern) && name.endsWith(".car");
+                    }
+                });
+
+                // Remove deployed CApps which has downloaded CApp name prefix
+                if (existingCApps != null && existingCApps.length != 0) {
+                    for (File cApp : existingCApps) {
+                        Files.delete(cApp.toPath());
+                        log.info(cApp.getName() + " file deleted from " + cAppsDirectoryPath + " directory");
+                    }
+                    jsonResponse.put(Constants.MESSAGE_JSON_ATTRIBUTE, "Successfully removed Carbon Application(s) " +
+                            "named " + cAppNamePattern);
+                } else {
+                    jsonResponse = Utils.createJsonError("Carbon Application(s) named or patterned " +
```

Comment:
I think we need to remove the pattern idea from everywhere in the delete path

Reply:
yes correct variable name need to be changed